data/content_scripts/posting_process/sendkeys.js

```diff
@@ -0,0 +1,766 @@
+// Cross-broswer implementation of text ranges and selections
+// documentation: http://bililite.com/blog/2011/01/17/cross-browser-text-ranges-and-selections/
+// Version: 2.6
+// Copyright (c) 2013 Daniel Wachsstock
+// MIT license:
```

Comment:
The library is only needed when posting a link, so you can add the library
when starting the posting process. This is a rare event so you can inject
it into a specific tab's scripting environment and all of its of iframes.
When the user navigates the tab the library will automatically be unloaded
from that tab, no? Did this clear things up? Thanks for asking the pointed
questions :)

(sent from my phone)